server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/routes/MailQueueRoutes.java

```diff
@@ -0,0 +1,57 @@
+package org.apache.james.webadmin.routes;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+
+import org.apache.james.queue.api.MailQueueFactory;
+import org.apache.james.queue.api.ManageableMailQueue;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.Service;
+
+public class MailQueueRoutes implements Routes {
+
+    @VisibleForTesting  static final String BASE_URL = "/mailQueues";
+    private final MailQueueFactory<ManageableMailQueue> mailQueueFactory;
+    private final JsonTransformer jsonTransformer;
+
+    @Inject
+    @VisibleForTesting MailQueueRoutes(MailQueueFactory<ManageableMailQueue> mailQueueFactory, JsonTransformer jsonTransformer) {
+        this.mailQueueFactory = mailQueueFactory;
+        this.jsonTransformer = jsonTransformer;
+    }
+
+    @Override
+    public void define(Service service) {
+        defineListQueues(service);
+    }
+
+    @GET
+    @ApiOperation(
+        value = "Listing existing MailQueues"
+    )
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.OK_200, message = "OK", response = List.class),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    public void defineListQueues(Service service) {
+        service.get(BASE_URL,
+            (request, response) ->
+                mailQueueFactory
+                    .listCreatedMailQueues()
+                    .stream()
+                    .map(ManageableMailQueue::getName)
```

Comment:
I'm not super fan of just having a "getName" as part of the payload. 

This makes future extensions harder. For instance if you want to add more information to what a queue is. If you choose an object then you : 
 - Have a more descriptive output (the client understands the value correspond to the name)
 - Extension and field addition can be done without changing the client. 

I would thus propose a payload being:

```
[
   {"name": "spool"},
   {"name": "outgoing"}
]
```

Rather than:

```
["name", "outgoing"]
```

Reply:
Not OK. In our REST APIs, we rather returned the id in the list and details in the GET on the entity.
It makes more sense to me.

Reply:
I agree with Matthieu, reverted.